File: product-processor-service/src/main/java/com/example/product_processor_service/service/common/kafka/listener/KafkaListenerService.java
package com.example.product_processor_service.service.common.kafka.listener;

import org.apache.kafka.clients.consumer.ConsumerRecord;

import java.io.Serializable;

public interface KafkaListenerService <K extends Serializable, V extends Serializable> {
    void listen(ConsumerRecord<K, V> listenableObject);
}

File: email-sender-service/src/main/kotlin/com/example/email_sender_service/service/common/kafka/listener/KafkaListenerService.kt
package com.example.email_sender_service.service.common.kafka.listener

import org.apache.kafka.clients.consumer.ConsumerRecord
import java.io.Serializable

interface KafkaListenerService<K : Serializable, V : Serializable> {
    fun listen(consumerRecord: ConsumerRecord<K, V>)
}